bin/ghe-host-check

```diff
@@ -15,32 +15,57 @@ set -e
 
 while true; do
   case "$1" in
-    -h|--help)
-      export GHE_SHOW_HELP=true
-      shift
-      ;;
-    --version)
-      export GHE_SHOW_VERSION=true
-      shift
-      ;;
-    -*)
-      echo "Error: invalid argument: '$1'" 1>&2
-      exit 1
-      ;;
-    *)
-      break
-      ;;
+  -h | --help)
+    export GHE_SHOW_HELP=true
+    shift
+    ;;
+  --version)
+    export GHE_SHOW_VERSION=true
+    shift
+    ;;
+  -*)
+    echo "Error: invalid argument: '$1'" 1>&2
+    exit 1
+    ;;
+  *)
+    break
+    ;;
   esac
 done
 
 # Bring in the backup configuration
 # shellcheck source=share/github-backup-utils/ghe-backup-config
-. "$( dirname "${BASH_SOURCE[0]}" )/../share/github-backup-utils/ghe-backup-config"
+. "$(dirname "${BASH_SOURCE[0]}")/../share/github-backup-utils/ghe-backup-config"
 
 # Use the host provided on the command line if provided, or fallback on the
 # $GHE_HOSTNAME configured in backup.config when not present.
 host="${1:-$GHE_HOSTNAME}"
 
+CLUSTER=false
+if ghe-ssh "$host" -- \
+  "[ -f '$GHE_REMOTE_ROOT_DIR/etc/github/cluster' ]"; then
+  CLUSTER=true
+fi
+
+if "$CLUSTER"; then
+  node_version_list=$(ghe-ssh "$host" ghe-cluster-each -- ghe-version)
+  echo "$node_version_list" 1>&2
+  first_node_version=""
+  (
+    IFS=$'\n'
+    for node_version in $node_version_list; do
+      if [ -z "$first_node_version" ]; then
+        first_node_version=$(echo "$node_version" | cut -f2 -d":")
+      fi
+      current_node_version=$(echo "$node_version" | cut -f2 -d":")
+      if [ "$first_node_version" != "$current_node_version" ]; then
+        echo "Error: Not all nodes are running the same version! Please ensure all nodes are running the same version before using backup-utils." 1>&2
+        exit 1
+      fi
+    done
+  )
```

Comment:
A bash-ier option (which is not always preferable) could be something like this:

```suggestion
  distinct_versions=$(echo "$node_version_list" | awk '{split($0, a, ":"); print a[2]}' | uniq | wc -l)
  if [ $distinct_versions -ne 1 ]; then
    echo "Error: Not all nodes are running the same version! Please ensure all nodes are running the same version before using backup-utils." 1>&2
    exit 1
  fi
```

The `awk` command is splitting each line on `:` and storing the second half in an array named `a`